### DISCUSSION TITLE
Is Pixi Cache thread-safe?

### DISCUSSION THREAD

**Discussion by cc636489** [prefix-dev/pixi#2745] [Q&A] [answered]
Hi everyone,

I’ve recently started using Pixi for managing Python environments and packages. While exploring its features, I couldn't find detailed information about its cache mechanism. Specifically, I’d like to know:

- Is the Pixi cache thread-safe?
- If it is, does it implement any locking mechanism to prevent simultaneous writes or reads from corrupting the cache?

Understanding this is crucial for using Pixi in multi-threaded or concurrent processes. Any insights, documentation links, or user experiences would be greatly appreciated!

Thanks in advance!

**Comment by baszalmstra** [comment, marked as answer]
Yes is thread and multi-process safe. We test this in CI a lot. We use file locks in the several places to make sure this works.

**Reply by cc636489** [reply]
Thanks for your input, that helps my understanding!
